vendor/runner/cli.js

```diff
@@ -0,0 +1,101 @@
+// Usage:
+//
+// Run all vendors:
+//   npm run test:vendor
+//
+// Run some vendors:
+//   npm run test:vendor -- --only=vendor1,vendor2,vendor3
+//
+// Infer the current PR from the Travis environment, and look for bracketed,
+// space-separated vendor names in the pull request title. If none are found,
+// do not run any tests. For example:
+// Pull request title: [travis sonar] Support user token authentication
+//   npm run test:vendor -- --pr
+// is equivalent to
+//   npm run test:vendor -- --only=travis,sonar
+
+'use strict';
+
+const request = require('request');
+const minimist = require('minimist');
+const difference = require('lodash.difference');
+const serverHelpers = require('../../test/in-process-server-helpers');
+const Runner = require('./runner');
+
+const getTitle = (repoSlug, pullRequest) => new Promise((resolve, reject) => {
+  const options = {
+    uri: `https://api.github.com/repos/${repoSlug}/pulls/${pullRequest}`,
+    json: true,
+    headers: { 'User-Agent': 'badges/shields' },
+  };
+  request(options, (err, res, json) => {
```

Comment:
If I use `node-fetch` I'm adding a whole interface that devs wouldn't be used to. If I use `request` and the person doesn't understand promises they'll be confused, and of course it would be better generally to use a wrapper. If I write new code using callbacks, that's sad. Three lousy options. I settled on wrapping `request` because it's used everywhere else in the project, and this github business is among the least important code, so under the circumstances it seemed the least bad. 😜 

Incidentally, I am not a huge fan of request's interface. Late in the callbacks era at my last job I wrote [this shim](https://github.com/bodylabs/request-shim-for-callbacks#usage), not to go bonkers copying and pasting line after line of boilerplate error handling. So, I probably would not choose a request wrapper, though I'm fine with `node-fetch`, or just de-promisifying the thing.